### ISSUE TITLE
ValueError: Failed to parse time "UNLIMITED"

### ISSUE THREAD

**Issue by prerakmody** [troycomi/reportseff#13]
Upon running the command `reportseff -u <myusername>`, I run into a ValueError.

```
Error processing entry: {'AdminComment': '', 'AllocCPUS': '4', 'Elapsed': '03:22:47', 'JobID': '11741520', 'JobIDRaw': '11741520', 'MaxRSS': '', 'NNodes': '1', 'REQMEM': '7Gc', 'State': 'COMPLETED', 'Timelimit': 'UNLIMITED', 'TotalCPU': '01:15:11'}
Traceback (most recent call last):
  File "/exports/lkeb-hpc/ppmody/anaconda3/envs/hecktor2021/bin/reportseff", line 8, in <module>
    sys.exit(main())
  File "/exports/lkeb-hpc/ppmody/anaconda3/envs/hecktor2021/lib/python3.8/site-packages/click/core.py", line 1130, in __call__
    return self.main(*args, **kwargs)
  File "/exports/lkeb-hpc/ppmody/anaconda3/envs/hecktor2021/lib/python3.8/site-packages/click/core.py", line 1055, in main
    rv = self.invoke(ctx)
  File "/exports/lkeb-hpc/ppmody/anaconda3/envs/hecktor2021/lib/python3.8/site-packages/click/core.py", line 1404, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/exports/lkeb-hpc/ppmody/anaconda3/envs/hecktor2021/lib/python3.8/site-packages/click/core.py", line 760, in invoke
    return __callback(*args, **kwargs)
  File "/exports/lkeb-hpc/ppmody/anaconda3/envs/hecktor2021/lib/python3.8/site-packages/reportseff/console.py", line 97, in main
    output, entries = get_jobs(args)
  File "/exports/lkeb-hpc/ppmody/anaconda3/envs/hecktor2021/lib/python3.8/site-packages/reportseff/console.py", line 148, in get_jobs
    raise error
  File "/exports/lkeb-hpc/ppmody/anaconda3/envs/hecktor2021/lib/python3.8/site-packages/reportseff/console.py", line 145, in get_jobs
    job_collection.process_entry(entry, add_job=add_jobs)
  File "/exports/lkeb-hpc/ppmody/anaconda3/envs/hecktor2021/lib/python3.8/site-packages/reportseff/job_collection.py", line 164, in process_entry
    self.jobs[job_id].update(entry)
  File "/exports/lkeb-hpc/ppmody/anaconda3/envs/hecktor2021/lib/python3.8/site-packages/reportseff/job.py", line 110, in update
    self._update_main_job(entry)
  File "/exports/lkeb-hpc/ppmody/anaconda3/envs/hecktor2021/lib/python3.8/site-packages/reportseff/job.py", line 130, in _update_main_job
    _parse_slurm_timedelta(entry["Timelimit"]) if "Timelimit" in entry else 1
  File "/exports/lkeb-hpc/ppmody/anaconda3/envs/hecktor2021/lib/python3.8/site-packages/reportseff/job.py", line 337, in _parse_slurm_timedelta
    raise ValueError(f'Failed to parse time "{delta}"')
ValueError: Failed to parse time "UNLIMITED"
```

---

It does work when called for specific jobs `reportseff -p 11772368`
```
JobID|State|Elapsed|TimeEff|CPUEff|MemEff
11743123|OUT_OF_MEMORY|1-06:32:09|16.2|37.2|94.3
```
Though I dont see colors and arranged columns.

**Comment by troycomi** [comment]
Just to confirm, does `reportseff -p 11741520` give the same error?

It seems the offending job has no time limit. As a user, would reporting the time efficiency as `---` be ok?

**Comment by prerakmody** [comment]
So it fails when I run `reportseff -p 11741520`

```
Error processing entry: {'AdminComment': '', 'AllocCPUS': '4', 'Elapsed': '03:22:47', 'JobID': '11741520', 'JobIDRaw': '11741520', 'MaxRSS': '', 'NNodes': '1', 'REQMEM': '7Gc', 'State': 'COMPLETED', 'Timelimit': 'UNLIMITED', 'TotalCPU': '01:15:11'}
Traceback (most recent call last):
  File "/exports/lkeb-hpc/ppmody/anaconda3/envs/hecktor2021/bin/reportseff", line 8, in <module>
    sys.exit(main())
  File "/exports/lkeb-hpc/ppmody/anaconda3/envs/hecktor2021/lib/python3.8/site-packages/click/core.py", line 1130, in __call__
    return self.main(*args, **kwargs)
  File "/exports/lkeb-hpc/ppmody/anaconda3/envs/hecktor2021/lib/python3.8/site-packages/click/core.py", line 1055, in main
    rv = self.invoke(ctx)
  File "/exports/lkeb-hpc/ppmody/anaconda3/envs/hecktor2021/lib/python3.8/site-packages/click/core.py", line 1404, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/exports/lkeb-hpc/ppmody/anaconda3/envs/hecktor2021/lib/python3.8/site-packages/click/core.py", line 760, in invoke
    return __callback(*args, **kwargs)
  File "/exports/lkeb-hpc/ppmody/anaconda3/envs/hecktor2021/lib/python3.8/site-packages/reportseff/console.py", line 97, in main
    output, entries = get_jobs(args)
  File "/exports/lkeb-hpc/ppmody/anaconda3/envs/hecktor2021/lib/python3.8/site-packages/reportseff/console.py", line 148, in get_jobs
    raise error
  File "/exports/lkeb-hpc/ppmody/anaconda3/envs/hecktor2021/lib/python3.8/site-packages/reportseff/console.py", line 145, in get_jobs
    job_collection.process_entry(entry, add_job=add_jobs)
  File "/exports/lkeb-hpc/ppmody/anaconda3/envs/hecktor2021/lib/python3.8/site-packages/reportseff/job_collection.py", line 164, in process_entry
    self.jobs[job_id].update(entry)
  File "/exports/lkeb-hpc/ppmody/anaconda3/envs/hecktor2021/lib/python3.8/site-packages/reportseff/job.py", line 110, in update
    self._update_main_job(entry)
  File "/exports/lkeb-hpc/ppmody/anaconda3/envs/hecktor2021/lib/python3.8/site-packages/reportseff/job.py", line 130, in _update_main_job
    _parse_slurm_timedelta(entry["Timelimit"]) if "Timelimit" in entry else 1
  File "/exports/lkeb-hpc/ppmody/anaconda3/envs/hecktor2021/lib/python3.8/site-packages/reportseff/job.py", line 337, in _parse_slurm_timedelta
    raise ValueError(f'Failed to parse time "{delta}"')
ValueError: Failed to parse time "UNLIMITED"
```

And I guess, since a specific time was not given to the job, we can display `---` for the `TimeEff` column.

**Comment by troycomi** [comment]
Should be fixed with v2.2.3.  Please reopen if you still have problems!
